### ISSUE TITLE
AppImage build seems corrupted

### ISSUE THREAD

**Issue by mscharley** [electron-userland/electron-builder#6624]
* **Electron-Builder Version**: 23.0.0-alpha.2

* **Node Version**: 16.13.1

* **Electron Version**: 17.0.0
* **Electron Type (current, beta, nightly)**: current

* **Target**: AppImage

I'm using the alpha version because I was running into #6563 .

This seems to be an issue with electron 17.x, if I roll back to 16.0.8 then it builds and runs fine. I'm building on Arch Linux targeting AppImage. The build runs fine, but when I try to run it I get the following output:

```
$ electron-builder --publish never

  • electron-builder  version=23.0.0-alpha.2 os=5.15.19-1-lts
  • loaded configuration  file=/home/mscharley/Code/mscharley/app/electron-builder.yml
  • writing effective config  file=dist/builder-effective-config.yaml
  • packaging       platform=linux arch=x64 electron=17.0.0 appOutDir=dist/linux-unpacked
  • building        target=AppImage arch=x64 file=dist/app-0.1.0.AppImage
  • default Electron icon is used  reason=application icon is not set

$ ./dist/app-0.1.0.AppImage
Squashfs image uses (null) compression, this version supports only xz, zlib.

Cannot mount AppImage, please check your FUSE setup.
You might still be able to extract the contents of this AppImage
if you run it with the --appimage-extract option.
See https://github.com/AppImage/AppImageKit/wiki/FUSE
for more information
open dir error: No such file or directory
```

**Comment by mmaietta** [comment]
Could you please post your snap configuration for electron-builder?

We upgraded to enable `lzo` compression for snap as the default in https://github.com/electron-userland/electron-builder/pull/6201
In your snap options, try setting `compression: 'xz'`

**Comment by mscharley** [comment]
No change with the compression option added.

My `build-effective-config.yaml`:

```yaml
directories:
  output: dist
  buildResources: build
appId: com.github.mscharley.app
files:
  - filter:
      - '**/*'
      - '!**/node_modules/*/{CHANGELOG.md,README.md,README,readme.md,readme}'
      - '!**/node_modules/*/{test,__tests__,tests,powered-test,example,examples}'
      - '!**/node_modules/*.d.ts'
      - '!**/node_modules/.bin'
      - '!**/*.{iml,o,hprof,orig,pyc,pyo,rbc,swp,csproj,sln,xproj}'
      - '!.editorconfig'
      - '!**/._*'
      - '!**/{.DS_Store,.git,.hg,.svn,CVS,RCS,SCCS,.gitignore,.gitattributes}'
      - '!**/{__pycache__,thumbs.db,.flowconfig,.idea,.vs,.nyc_output}'
      - '!**/{appveyor.yml,.travis.yml,circle.yml}'
      - '!**/{npm-debug.log,yarn.lock,.yarn-integrity,.yarn-metadata.json}'
linux:
  target: AppImage
  category: Utility
snap:
  compression: "xz"
publish:
  - github
electronVersion: 17.0.0
```

**Comment by mmaietta** [comment]
Whoops. Maybe `compression: xz` is supposed to be under `linux`? It's a stab in the dark as I haven't used it before

**Comment by mscharley** [comment]
I tried under both `linux` and `appImage` top level and they both failed the configuration validation and didn't even try to build. Based on the PR you linked above, that's a feature of snap, why would it apply to appimages?

**Comment by mscharley** [comment]
Still an issue with the final release of electron-builder 23.0.0 when building against either electron 17.0.1 or 16.0.9.

**Comment by mscharley** [comment]
Adding this to the top level gets a working build as it triggers xz encryption:

```
compression: maximum
```

But the resulting build is prohibitively slow to run. Neither of the other two valid top level values (`normal` and `store`) produce a valid build.

**Comment by mmaietta** [comment]
I'm wondering if compression is no longer optional param. Could you give this patch a try using [patch-package](https://www.npmjs.com/package/patch-package)? That should unblock you for the interim

`app-builder-lib+23.0.1.patch`
```
diff --git a/node_modules/app-builder-lib/out/targets/AppImageTarget.js b/node_modules/app-builder-lib/out/targets/AppImageTarget.js
index 31344b3..0be0fe4 100644
--- a/node_modules/app-builder-lib/out/targets/AppImageTarget.js
+++ b/node_modules/app-builder-lib/out/targets/AppImageTarget.js
@@ -75,13 +75,12 @@ class AppImageTarget extends core_1.Target {
                 fileAssociations: this.packager.fileAssociations,
                 ...options,
             }),
+            "--compression",
+            packager.compression === "maximum" ? "xz" : "zlib",
         ];
         appBuilder_1.objectToArgs(args, {
             license,
         });
-        if (packager.compression === "maximum") {
-            args.push("--compression", "xz");
-        }
         await packager.info.callArtifactBuildCompleted({
             file: artifactPath,
             safeArtifactName: packager.computeSafeArtifactName(artifactName, "AppImage", arch, false),
```

**Comment by mscharley** [comment]
@mmaietta using that patch: 

```
> electron-builder --publish never

  • electron-builder  version=23.0.1 os=5.15.24-1-lts
  • loaded configuration  file=/home/mscharley/Code/mscharley/notes-nc/electron-builder.yml
  • writing effective config  file=dist/builder-effective-config.yaml
  • packaging       platform=linux arch=x64 electron=17.1.0 appOutDir=dist/linux-unpacked
  • downloading     url=https://github.com/electron/electron/releases/download/v17.1.0/electron-v17.1.0-linux-x64.zip size=99 MB parts=8
  • downloaded      url=https://github.com/electron/electron/releases/download/v17.1.0/electron-v17.1.0-linux-x64.zip duration=10.948s
  • building        target=AppImage arch=x64 file=dist/notes-nc-0.1.0.AppImage
  • default Electron icon is used  reason=application icon is not set
  ⨯ enum value must be one of xz,gzip, got 'zlib'
  ⨯ /home/mscharley/Code/mscharley/notes-nc/node_modules/app-builder-bin/linux/x64/app-builder process failed ERR_ELECTRON_BUILDER_CANNOT_EXECUTE
Exit code:
1  failedTask=build stackTrace=Error: /home/mscharley/Code/mscharley/notes-nc/node_modules/app-builder-bin/linux/x64/app-builder process failed ERR_ELECTRON_BUILDER_CANNOT_EXECUTE
```

See my comment above where I already tried various values for the encryption option. It looks like `app-builder-lib` and `app-builder-bin` seem to disagree on what the valid values for that option are.

**Comment by mmaietta** [comment]
Closing as duplicate of #6678 to consolidate convos

**Comment by Slapbox** [comment]
@mmaietta I think you forgot to hit close on that comment.
